File: server/internal/dto/response/ticket.go
// Package response 定义申告管理相关响应 DTO。
//
// 与 TECH.md §5.2 申告 API 端点对齐。
// 列表响应包含分页信息，详情响应包含提交人信息和处理记录时间线。
package response

// TicketListResponse 申告列表响应（分页）。
type TicketListResponse struct {
	Tickets []TicketItem `json:"items"`
	Total   int64        `json:"total"`
}

// TicketItem 列表中的申告条目（精简字段）。
type TicketItem struct {
	ID              int64  `json:"id"`
	TicketNo        string `json:"ticket_no"`
	UserID          int64  `json:"user_id"`
	SubmitterName   string `json:"submitter_name"`
	Title           string `json:"title"`
	Urgency         int16  `json:"urgency"`
	ImpactScope     int16  `json:"impact_scope"`
	ContactPhone    string `json:"contact_phone"`
	Status          int16  `json:"status"`
	StatusText      string `json:"status_text"`
	SupplementCount int16  `json:"supplement_count"`
	CreatedAt       string `json:"created_at"`
	UpdatedAt       string `json:"updated_at"`
}

// TicketDetailResponse 申告详情响应（含提交人信息和处理记录时间线）。
type TicketDetailResponse struct {
	TicketItem
	Description     string              `json:"description"`
	AffectedSystems []string            `json:"affected_systems"`
	ContactEmail    string              `json:"contact_email"`
	Source          int16               `json:"source"`
	Records         []TicketRecordItem  `json:"records"`
}

// TicketRecordItem 处理记录条目。
type TicketRecordItem struct {
	ID         int64  `json:"id"`
	TicketID   int64  `json:"ticket_id"`
	OperatorID int64  `json:"operator_id"`
	Action     string `json:"action"`
	Content    string `json:"content"`
	Detail     string `json:"detail"`
	CreatedAt  string `json:"created_at"`
}

File: web/src/api/ticket.ts
/**
 * 申告 API 封装（门户端）
 *
 * 提供申告创建、列表查询、详情查询、补充信息等门户端接口。
 * 后台管理端申告接口（列表管理、状态操作）在 admin API 中独立封装。
 *
 * TODO(api/ticket): TicketRecord.detail 使用 Record<string, any> — any 应替换为具体类型。
 * TODO(api/ticket): listMyTickets/getTicketDetail 响应类型缺少 ApiResponse<T> 包装，
 *                   TicketListResponse/TicketDetail 无法正确匹配后端 { code, message, data } 结构。
 * TODO(api/ticket): createTicket/supplementTicket 无泛型参数，返回 any — 应补充类型。
 */
import request from '../utils/request'

// =============================================================================
// 类型定义
// =============================================================================

export interface CreateTicketParams {
  title: string
  description: string
  urgency: number          // 1=低, 2=中, 3=高
  impact_scope?: number    // 1=个人, 2=部门, 3=全公司
  affected_systems?: string[]
  contact_phone: string
  contact_email?: string
  chat_context?: string    // JSON 字符串，从问答转申告时带入
}

export interface SupplementTicketParams {
  content: string
}

export interface TicketItem {
  id: number
  ticket_no: string
  user_id: number
  submitter_name: string
  title: string
  urgency: number
  impact_scope: number
  contact_phone: string
  status: number
  status_text: string
  supplement_count: number
  created_at: string
  updated_at: string
}

export interface TicketRecord {
  id: number
  ticket_id: number
  operator_id: number
  action: string
  content: string
  detail?: Record<string, any>
  created_at: string
}

export interface TicketDetail {
  id: number
  ticket_no: string
  user_id: number
  submitter_name: string
  title: string
  description: string
  urgency: number
  impact_scope: number
  affected_systems: string[]
  contact_phone: string
  contact_email: string
  status: number
  status_text: string
  supplement_count: number
  chat_context?: string
  source: number
  records: TicketRecord[]
  created_at: string
  updated_at: string
}

export interface TicketListResponse {
  items: TicketItem[]
  total: number
}

// =============================================================================
// API 方法
// =============================================================================

/** 创建申告 */
export function createTicket(data: CreateTicketParams) {
  return request.post('/api/v1/portal/tickets', data)
}

/** 查询当前用户的申告列表 */
export function listMyTickets(page: number = 1, pageSize: number = 10) {
  return request.get<TicketListResponse>('/api/v1/portal/tickets', {
    params: { page, page_size: pageSize }
  })
}

/** 查询申告详情 */
export function getTicketDetail(id: number) {
  return request.get<TicketDetail>(`/api/v1/portal/tickets/${id}`)
}

/** 补充申告信息（仅"需补充信息"状态可操作） */
export function supplementTicket(id: number, data: SupplementTicketParams) {
  return request.post(`/api/v1/portal/tickets/${id}/supplement`, data)
}
